lib/collection/src/operations/config_diff.rs

```diff
@@ -85,6 +86,79 @@ pub struct HnswConfigDiff {
     pub payload_m: Option<usize>,
 }
 
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Validate, Clone, Merge)]
+pub struct StrictModeConfig {
+    // Global
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub enabled: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub max_filter_limit: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub max_timeout: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub unindexed_filtering_retrieve: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub unindexed_filtering_update: Option<bool>,
+
+    // Search
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub search_max_hnsw_ef: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub search_allow_exact: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub search_max_oversampling: Option<f64>,
+
+    // Group-by
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub group_by_max_group_size: Option<usize>,
+
+    // Recommend
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub recommend_max_examples: Option<usize>,
```

Comment:
Recommend for average vector with multiple examples is not too expensive, does this apply only for `best_score`? If it is the case, name should be something like: `recommend_best_score_max_inputs`

Reply:
CC @generall

Reply:
I just discussed this with @generall and he prefers to keep the amount of parameters low. 
To address that, I've merged `recommend_max_examples` and `discovery_max_context_size` into `max_input_examples` to make it more generic.